src/util/util.js

```diff
@@ -124,6 +124,29 @@ function extractURLsFromTemplates(URLTemplates) {
   return URLTemplates;
 }
 
+/**
+ * Filter URLTemplates elements to keep only valid and safe URL templates.
+ *   To be valid, urls should:
+ *   - have the same protocol as the client
+ *   or
+ *   - be protocol-relative urls
+ *
+ * @param {String|String[]|Object[]} URLTemplates - A string|string[]|object[] of url templates.
+ */
+function filterValidUrlTemplates(URLTemplates) {
+  if (Array.isArray(URLTemplates)) {
+    return URLTemplates.filter(urlTemplate => {
+      const url = urlTemplate.hasOwnProperty('url') ? urlTemplate.url : urlTemplate;
+      return isValidUrl(url);
+    })
+  }
+  return isValidUrl(URLTemplates);
+}
+
+function isValidUrl(url) {
+  return (url?.startsWith('http://') || url?.startsWith('https://') || url?.startsWith('//'));
```

Comment:
Isn't it more performant and also more elegant with only one regex?

Reply:
Not sure, because this can make http url valid for https hosts, and I'm pretty bad at regex xD.

My first version used window.location but the testing can be limited if we mock the location to https://something for example.
`return (url?.startsWith(window.location.protocol) || url?.startsWith('//'))`